### ISSUE TITLE
Lesson Contribution: Setup Information

### ISSUE THREAD

**Issue by Talishask** [datacarpentry/genomics-r-intro#97]
I'm a member of The Carpentries staff and I'm submitting this issue on behalf of another member of the community. In most cases, I won't be able to follow up or provide more details other than what I'm providing below.

---

I've found that the setup page for the Intro to R and RStudio for Genomics has not been created yet, and the setup requirements for the Genomics Workshop Overview seem to differ somewhat as the former lesson does not require the bulk of the listed programs (including spreadsheet software) nor most of the data used in the Genomics Workshop. Reading through the overall lesson, it appears that the requirements for the Intro to R and RStudio for Genomics workshop are: R; RStudio; the dplyr and ggplot2 packages; and the "combined_tidy_vcf.csv" file listed on the associated figshare page for the Genomics Workshop. Additionally, there is one Excel spreadsheet (Ecoli_metadata.xlsx; located at /home/dcuser/dc_sample_data/R on the Amazon Web Services instance) that is used in Section 03 under the heading "Importing data from Excel" (approx. 80% of the way down the page) that does not appear to be available otherwise. With that in mind, I have gone through the lessons for the Intro to R and RStudio for Genomics workshop and would like to submit the following for the Setup page for this lesson; please note that I am not familiar with GitHub, nor am I familiar with how to use the templates for these pages, so the following is largely unformatted text based off of the setup pages for the Genomics Workshop Overview and the Social Science Workshop Overview. I have included hyperlinks similar to the other setup pages and attempted some formatting to indicate headings (or other information) where relevant.
---

**Overview**

This workshop is designed to be run using RStudio Server on pre-imaged Amazon Web Services (AWS) instances. All of the software and data used in the workshop are hosted on an Amazon Machine Image (AMI). Please follow the instructions below to prepare your computer for the workshop:

- Option A (AWS)
OR
- Option B (Local machine)

### Option A (Recommended): Using the lessons with Amazon Web Services (AWS)

If you are signed up to take a Genomics Data Carpentry workshop, you do not need to worry about setting up an AMI instance. The Carpentries staff will create an instance for you and this will be provided to you at no cost. This is true for both self-organized and centrally-organized workshops. Your Instructor will provide instructions for connecting to the AMI instance at the workshop.

If you would like to work through these lessons independently, outside of a workshop, you will need to start your own AMI instance. Follow [these instructions on creating an Amazon instance](https://datacarpentry.org/genomics-workshop/AMI-setup/). Use the AMI ami-04b3bc83255f918b0 (Data Carpentry Genomics with R 4.0) listed on the Community AMIs page. Please note that you must set your location as N. Virginia in order to access this community AMI. You can change your location in the upper right corner of the main AWS menu bar. The cost of using this AMI for a few days, with the t2.medium instance type is very low (about USD $1.50 per user, per day). Data Carpentry has no control over AWS pricing structure and provides this cost estimate with no guarantees. Please read AWS documentation on pricing for up-to-date information.

If you're an Instructor or Maintainer or want to contribute to these lessons, please get in touch with us team@carpentries.org and we will start instances for you.

### Option B: Using the lessons on your local machine

While not recommended, it is possible to work through the lessons on your local machine (i.e. without using AWS). To do this, you will need to install all of the software used in the workshop and obtain a copy of the dataset. Instructions for doing this are below.

**Data**
The data used in this workshop is available on FigShare. You can download the necessary data used in this workshop by clicking [this download link](https://ndownloader.figshare.com/files/14632895). The file is 168.17 KB.

Clicking the download link will automatically download the file to your default download directory as a single spreadsheet (.csv) file.

Please read the FigShare page linked below for information about the data and access to the data files.

[FigShare Data Carpentry Genomics Beta 2.0](https://figshare.com/articles/Data_Carpentry_Genomics_beta_2_0/7726454)

Software
****Software | Install | Manual | Available for
-- | -- | -- | --
R | See install instructions below |   | Linux, MacOS, Windows
RStudio | Link | Cheatsheet | Linux, MacOS, Windows

**R and RStudio**

•	R and RStudio are separate downloads and installations. R is the underlying statistical computing environment, but using R alone is no fun. RStudio is a graphical integrated development environment (IDE) that makes using R much easier and more interactive. You need to install R before you install RStudio. After installing both programs, you will need to install a specific R package within RStudio. Follow the instructions below for your operating system, and then follow the instructions for installing ggplot2. Additionally, we will also be using the dplyr package for R, although installation of this will be covered during the lesson.

_Windows_

If you already have R and RStudio installed

If you don't have R and RStudio installed

_macOS_

If you already have R and RStudio installed

If you don't have R and RStudio installed

_Linux_

•	Follow the instructions for your distribution from [CRAN](https://cloud.r-project.org/bin/linux), they provide information to get the most recent version of R for common distributions. For most distributions, you could use your package manager (e.g., for Debian/Ubuntu run sudo apt-get install r-base, and for Fedora sudo yum install R), but we don't recommend this approach as the versions provided by this are usually out of date. In any case, make sure you have at least R 4.0.
•	Go to the [RStudio download page](https://www.rstudio.com/products/rstudio/download/#download)
•	Under Installers select the version that matches your distribution, and install it with your preferred method (e.g., with Debian/Ubuntu sudo dpkg -i rstudio-x.yy.zzz-amd64.deb at the terminal).
•	Once it's installed, open RStudio to make sure it works and you don't get any error messages.

•	After installing R and RStudio, you need to install the [ggplot2 ](https://cran.r-project.org/web/packages/ggplot2/index.html) package. Start RStudio by double-clicking the icon and then type: install.packages("ggplot2"). You can also do this by going to Tools -> Install Packages and typing the names of the packages you want to install, separated by a comma.

**Comment by JasonJWilliamsNY** [comment]
Thanks @Talishask ! It's now on my plate to start fixing this. Do you want to actually submit this as a pull request so that you are properly credited in the GitHub contribution history? Let me know if I can further assist.

**Comment by Talishask** [comment]
Hi @JasonJWilliamsNY, I submitted this issue on behalf of an instructor training trainee and am unable to move this forward.

**Comment by JasonJWilliamsNY** [comment]
Ok did not realize. Do you have a name/GH ID for them so they can be credited for the contribution?

**Comment by Talishask** [comment]
I do not believe so. After giving them credit towards checkout I typically share the issue information with them so that they can follow the issue if they are on github. So unless they self identify I am not able to do more.

**Comment by JasonJWilliamsNY** [comment]
OK, thanks
